DigitalLearningSolutions.Web/Controllers/LearningPortalController/RecommendedLearningController.cs

```diff
@@ -0,0 +1,236 @@
+    using Microsoft.AspNetCore.Authorization;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.Extensions.Configuration;
+
+    [Authorize(Policy = CustomPolicies.UserOnly)]
+    [ServiceFilter(typeof(VerifyDelegateUserCanAccessSelfAssessment))]
+    public class RecommendedLearningController : Controller
+    {
+        private readonly IConfiguration configuration;
+        private readonly IFilteredApiHelperService filteredApiHelperService;
+        private readonly ISelfAssessmentService selfAssessmentService;
+
+        public RecommendedLearningController(
+            IFilteredApiHelperService filteredApiHelperService,
+            ISelfAssessmentService selfAssessmentService,
+            IConfiguration configuration
+        )
+        {
+            this.filteredApiHelperService = filteredApiHelperService;
+            this.selfAssessmentService = selfAssessmentService;
+            this.configuration = configuration;
+        }
+
+        [Route("/LearningPortal/SelfAssessment/{selfAssessmentId:int}/Results")]
+        public async Task<IActionResult> SelfAssessmentResults(int selfAssessmentId)
+        {
+            var candidateId = User.GetCandidateIdKnownNotNull();
+            
+            selfAssessmentService.SetSubmittedDateNow(selfAssessmentId, candidateId);
+            selfAssessmentService.SetUpdatedFlag(selfAssessmentId, candidateId, false);
+
+            if (!configuration.IsSignpostingUsed())
+            {
+                await UpdateFilteredProfileAndGoalsForDelegate(selfAssessmentId, candidateId);
+                return RedirectToAction("FilteredDashboard", new { selfAssessmentId });
+            }
+
+            return RedirectToAction("RecommendedLearning", new { selfAssessmentId });
+        }
+
+        [Route("/LearningPortal/SelfAssessment/{selfAssessmentId:int}/RecommendedLearning")]
+        public async Task<IActionResult> RecommendedLearning(int selfAssessmentId)
+        {
+            if (!configuration.IsSignpostingUsed())
+            {
+                return RedirectToAction("FilteredDashboard", new { selfAssessmentId });
+            }
+
+            var candidateId = User.GetCandidateIdKnownNotNull();
+            string destUrl = "/LearningPortal/SelfAssessment/" + selfAssessmentId + "/RecommendedLearning";
+            selfAssessmentService.SetBookmark(selfAssessmentId, candidateId, destUrl);
+            selfAssessmentService.UpdateLastAccessed(selfAssessmentId, candidateId);
+
+            return ReturnSignpostingRecommendedLearningView(selfAssessmentId, candidateId);
+        }
+
+        [Route("/LearningPortal/SelfAssessment/{selfAssessmentId:int}/Filtered/Dashboard")]
+        public async Task<IActionResult> FilteredDashboard(int selfAssessmentId)
+        {
+            if (configuration.IsSignpostingUsed())
+            {
+                return RedirectToActionPermanent("RecommendedLearning", new { selfAssessmentId });

```

Comment:
I fear the `Permanent` could cause trouble if we want to revert to Filtered temporarily for some reason. I think it's better to use a temporary redirect because users shouldn't end up going to this endpoint twice after we turn signposting on, unless they're doing it manually (they'll only go here because of the leftover bookmark).